src/cpp/src/marius.cpp

```diff
@@ -161,9 +160,14 @@ void marius_train(shared_ptr<MariusConfig> marius_config) {
                 evaluator->evaluate(false);
             }
         }
-    }
 
-    metadata.num_epochs = dataloader->epochs_processed_;
+
+        metadata.num_epochs = dataloader->epochs_processed_;
+        if(marius_config->training->checkpoint_after_epochs > 0 && (epoch + 1) % marius_config->training->checkpoint_after_epochs == 0 && 
```

Comment:
nit: there should be a space after the `if`

Also add a new line after the first `&&` so that each condition is on a new line. Makes it a bit more readable.